Support CEF112

kenhys opened the pull request:
# Which issue(s) this PR fixes:

Recreated version of #20.

# What this PR does / why we need it:

Currently, the adopted CEF version is 98.2.1, a bit older version.
The stable version is CEF 110 and the beta version is now CEF 111.

This PR follows up to a newer version.

![chronos110](https://user-images.githubusercontent.com/225841/223919015-357ec5cd-80eb-4d80-b984-08db1aa769ad.png)

![chronos111](https://user-images.githubusercontent.com/225841/223919027-9066609e-eeea-4ec9-ada7-a06871ee903e.png)

# How to verify the fixed issue:

1. Download the archive from GitHub action for stable and beta job.
2. Extract archives and run Chronos.exe
3. Check the behavior of Chronos.exe

## Expected result:

master branch is ready for CEF 110 and CEF 110.

kenhys committed:
Fix unknown PluginPolicy

It causes the following error such as: error C3646: 'PluginPolicy'

See also:

  Delete cef_web_plugin.h and plugin-related APIs (see issue #3047)

  https://bitbucket.org/chromiumembedded/cef/issues/3047/
  https://bitbucket.org/chromiumembedded/cef/commits/28c7f040016a

kenhys committed:
Follow  new PrintViewManagerBase::PrintToPdf

Migrate to changed  member variables

See
  Use new PrintViewManagerBase::PrintToPdf method (fixes issue #3377)

  https://bitbucket.org/chromiumembedded/cef/issues/3377/
  https://bitbucket.org/chromiumembedded/cef/commits/25c75c5fc4fc

kenhys committed:
Follow new ::Find implemention

See:

  alloy: Implement Find() using find_in_page::FindTabHelper (fixes
  issue #3098, see issue #3047)

  https://bitbucket.org/chromiumembedded/cef/issues/3098
  https://bitbucket.org/chromiumembedded/cef/commits/171d525aa42e

kenhys committed:
Assign popupFeatures explicitly

See also:

  Stop using CefStructBase for POD types (fixes issue #3357)

  https://bitbucket.org/chromiumembedded/cef/issues/3357/
  https://bitbucket.org/chromiumembedded/cef/commits/dad9bf87ef36

kenhys committed:
Disable CefRequestHandler::OnQuotaRequest

As CefRequestHandler::OnQuotaRequest was removed.

See also:
   https://crbug.com/1208141
   https://bitbucket.org/chromiumembedded/cef/commits/47d69a842a27384f9c42e59b9bd8bae3ab9e287c

kenhys committed:
github: unify with matrix build

This PR is follow up of https://github.com/ThinBridge/Chronos/pull/20/

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>

kenhys committed:
github: prepare Chronos.exe in artifact

kenhys committed:
Follow changes introduced with Chromium version 110.0.5481.0 (#1084008)

The following member variable was removed from _cef_popup_features_t.

 * menuBarVisible
 * statusBarVisible
 * toolBarVisible
 * scrollbarsVisible

The above variables are replaced with isPopup.
(derived from webKitFeatures.is_popup)

HashidaTKS commented on client_handler.h:
```diff
@@ -48,7 +48,10 @@ class ClientHandler : public CefClient,
-	typedef cef_plugin_policy_t PluginPolicy;
+#if CHROME_VERSION_MAJOR <= 98
+	// Since CEF99, cef_plugin_policy_t is not available anymore.
+	typedef cef_plugin_policy_t PluginPolicy;
```
Maybe this is a extra `CR` (`^M`).

kenhys replied:
@HashidaTKS Thanks, Fixed it.

HashidaTKS commented on .github/workflows/build.yaml:
```diff
@@ -5,10 +5,21 @@ on:
+      - if: matrix.version == 'stable'
+        name: Use stable version of CEF
+        run: |
+          bash .github/rewrite-cef-version.sh ${{ matrix.version }}
```
Maybe this comment is out of scope of this pull request, and just a question.

Do we need to rewrite `setup-cef.bat` ?
It seems that what we need to do is specifying `CEFVER` in `setup-cef.bat`.

So could we modify this as following?

* Set `CEFVER` from environment variables if specified in `setup-cef.bat`
  * Default: `cef_binary_111.2.1+g870da30+chromium-111.0.5563.64_windows32_minimal`
* Set only environment variable in `build.yaml` and don't rewrite  `setup-cef.bat`.

kenhys replied:
@HashidaTKS 

I don't know where `setup-cef.bat` comes from, so hesitate to modify it at that time.
It seems that giving CEFVER via the environment variable is straightforward now, so could you change in another PR, please? after that PR is merged,  this PR also shold follow it.

HashidaTKS replied:
> so could you change in another PR, please? 

Sure, I'll try it now.

HashidaTKS replied:
I have created a new PR for specifying CEFVER via the environment variable in `setup-cef.bat`
#32

HashidaTKS committed:
Add support for CEF 112

CefEnableHighDPISupport is removed as it is no longer needed.

https://github.com/chromiumembedded/cef/issues/3452

ashie commented:
We are now going to apply CEF112 for the next Chronos version.

HashidaTKS commented:
> after that PR is merged, this PR also should follow it.

We have not done that yet, but I have no objection to merge this.
We can follow to https://github.com/ThinBridge/Chronos/pull/32 as a follow up task in another PR.

ashie reviewed (approved):
LGTM for the first step of supporting CEF112.
We'll post additional pull request for the development of the next version.
